### PR TITLE
Erctl

### PR DESCRIPTION
**Pull request by haohanyuzmx**

### CODE REVIEW

File: pkg/agent/datapath/utils.go [hunk 1]
```diff
@@ -94,3 +97,44 @@ func ParseMacToUint64(b []byte) uint64 {
 	return uint64(b[5]) | uint64(b[4])<<8 | uint64(b[3])<<16 | uint64(b[2])<<24 |
 		uint64(b[1])<<32 | uint64(b[0])<<40 | 0<<48 | 0<<56
 }
+
+func datepathRule2RpcRule(entry *EveroutePolicyRuleEntry) *v1alpha1.RuleEntry {
```
Comment (KeyongSun) [inline pkg/agent/datapath/utils.go:101]: datepathRule2RpcRule -> datapath

File: pkg/erctl/cmd/show.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var showCmd = &cobra.Command{
+	Use:   "show",
+	Short: "specify which filed you want show",
+	Long: `all struct field has been set 'json:",omitempty"', so the programme will set those field null value
```
Comment (KeyongSun) [inline pkg/erctl/cmd/show.go:10]: Delete "all struct field has been set 'json:",omitempty"', so the programme will set those field null value"

File: pkg/apis/rule/v1alpha1/rule.proto [hunk 3]
```diff
@@ -0,0 +1,57 @@
+syntax = "proto3";
```
Comment (wu-cl) [inline pkg/apis/rule/v1alpha1/rule.proto:1]: move this part to `pkg/apis/rpc/v1alpha1`

File: pkg/erctl/cmd/rule.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	showCTflows                     bool
+)
+
+var ruleCmd = &cobra.Command{
+	Use:   "rule",
+	Short: "get RuleEntry from agent",
+	Long: `use grpc to get rules, default get all
+-f means get rule by flowids
+-r means get rule by ruleids
+sort rules by --srcip --dstip --dstport --protocol`,
```
Comment (wu-cl) [inline pkg/erctl/cmd/rule.go:24]: indent this part

File: pkg/erctl/cmd/show.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var showCmd = &cobra.Command{
+	Use:   "show",
+	Short: "specify which filed you want show",
+	Long: `[-D {fieldName...}] means don't show {fieldName...} (like blacklist)
+[-I {fieldName...}] means only show {fieldName...} (like whitelist)`,
```
Comment (wu-cl) [inline pkg/erctl/cmd/show.go:11]: indent

File: pkg/erctl/cmd/sort.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+)
+
+var sortCmd = &cobra.Command{
+	Use:   "sort",
+	Short: "filter rules by the way you specify",
+	Long: `provide [-d] difference set and [-i] intersection set
+For example:
+rule:[{foo:bar},{foo:aaa},{foo:bbb}]
+[-i foo=bar] return [{foo:bar}]
+[-d foo=bar] return [{foo:aaa},{foo:bbb}]`,
```
Comment (wu-cl) [inline pkg/erctl/cmd/sort.go:17]: indent

File: pkg/utils/utils.go [hunk 7]
```diff
@@ -161,3 +161,43 @@ func GenerateControllerID(typeID uint16) uint16 {
...
+	//
+	// No.1 1010 0010 1111 0001 0xA2F1  -  ovs register order
+	// No.2 1000 1111 0100 0101 0x8F45  -  ovs dpctl/dump-conntrack (left-right mirror from No.1)
+	// No.3 0100 0101 1000 1111 0x458F  -  netlink ct label
+	//
+	// label retrieve from netlink ct label, transfer it with little endian
+	// In the above case, it seems as No.2
+	// Since binary lib could only handle uint64, label (128 bits) split into TWO parts.
+	//
+	// The round number stores in high 10 bits. Here it means the right 10 bits in uint64 partA.
```
Comment (wu-cl) [inline pkg/utils/utils.go:176]: // The round number stores in high 4 bits. Here it means the right 4 bits in uint64 partA.
modify 10 bits to 4 bits, forgot change comment last time (￣.￣)

**Review by KeyongSun — APPROVED** [review]

**Pull request merged**